d2renderers/d2svg/d2svg.go

```diff
@@ -509,6 +509,91 @@ func defineShadowFilter(writer io.Writer) {
 </defs>`)
 }
 
+func render3dRect(targetShape d2target.Shape) string {
+	moveTo := func(p d2target.Point) string {
+		return fmt.Sprintf("M%d,%d ", p.X+targetShape.Pos.X, p.Y+targetShape.Pos.Y)
+	}
+	lineTo := func(p d2target.Point) string {
+		return fmt.Sprintf("L%d,%d ", p.X+targetShape.Pos.X, p.Y+targetShape.Pos.Y)
+	}
```

Comment:
these seem like generally reusable functions to have in the package scope

Reply:
I think the abstraction cost/benefit makes the most sense for how it is used here, the general use would also need to take in the targetShape as a parameter and each call becomes more verbose e.g. `lineTo(d2target.Point{...})` is nicer than `lineToInShape(targetShape, d2target.Point{...})`. If we need to do this elsewhere we can refactor at that point.